task/07-yield-tasks.js

```diff
@@ -108,7 +131,17 @@ function* depthTraversalTree(root) {
  *
  */
 function* breadthTraversalTree(root) {
-    throw new Error('Not implemented');
+    let nodes = [root];
+    let cond = true;
+    let i = 0;
+    while(cond) {
+        let node = nodes[i++];
+        if(node.children) {
+        nodes.push(...node.children);
+        };
+        cond = i != nodes.length;
```

Comment:
"operator !== is always preferable
sometimes ''!=' may cause an error
(check at console :
'0' != 0
and '0' !== 0)"